### PR TITLE
Carlos/c 389 chain specific deployment helper contracts

### PR DESCRIPTION
**Pull request by carlosdimatteo**
* add chain specific deployer library for mainnet and implemented in deployment script and tests
* make deployment exporter a library for better implementation
* cleanup duplicated code from tests and improv e fork handling
* move deployment loader to mainnet folder since it uses mainnet specific deployer library

### CODE REVIEW

**Comment by carlosdimatteo** [comment]
@artdgn regarding the deployment loader, i moved it into the mainnet folder, i thought maybe it would be good to prolly make one general and then chain specific inside the folder, but thought it might be overkill and not necessary , u let me know

File: script/arbitrum-mainnet/deployer.sol [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import { SetupHelper } from "../setup-helper.sol";
+import { DeploymentHelper } from "../deployment-helper.sol";
+
+library ArbitrumMainnetDeployer {
+    uint constant chainId = 42_161; // id for arbitrum mainnet
+    address constant USDC = 0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8;
+    address constant USDT = 0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9;
+    address constant WETH = 0x82aF49447D8a07e3bd95BD0d56f35241523fBab1;
+    address constant stETH = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84; // wsteth doesnt have any 0.3% fee with neither USDC nor USDT on arbitrum uniswap
+    address constant weETH = 0x35751007a407ca6FEFfE80b3cB397736D2cf4dbe; // Wrapped weETH on arbitrum doesnt have USDC pools on uniswap
```
Comment (artdgn) [inline script/arbitrum-mainnet/deployer.sol:15]: As comment mentions, this asset is irrelevant for now (doesn't have stable pools)

File: script/arbitrum-mainnet/deployer.sol [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    function deployAndSetupProtocol(address owner) internal returns (DeploymentResult memory result) {
+        require(chainId == block.chainid, "chainId does not match the chainId in config");
+
+        result.configHub = DeploymentHelper.deployConfigHub(owner);
+
+        address[] memory underlyings = new address[](5);
+        underlyings[0] = WETH;
+        underlyings[1] = WBTC;
+        underlyings[2] = MATIC;
+        underlyings[3] = weETH;
```
Comment (artdgn) [inline script/arbitrum-mainnet/deployer.sol:57]: as above, problematic assets

File: script/utils/deployment-exporter.s.sol [hunk 3]
```diff
@@ -127,95 +126,83 @@ contract DeploymentUtils is Script {
...
-        address configHubAddress = _parseAddress(parsedJson, ".configHub");
+        address configHubAddress = _parseAddress(vm, parsedJson, ".configHub");
 
         string[] memory allKeys = vm.parseJsonKeys(json, ".");
-        // Count valid asset pairs
         uint pairCount = 0;
         for (uint i = 0; i < allKeys.length; i++) {
             if (
-                bytes(allKeys[i]).length > 9 // exclude non asset pair keys
+                bytes(allKeys[i]).length > 9
```
Comment (artdgn) [inline script/utils/deployment-exporter.s.sol:155]: 9 should be probably be refactored

File: test/integration/arbitrum-mainnet/DeploymentLoader.sol [hunk 4]
```diff
@@ -30,18 +31,48 @@ abstract contract DeploymentLoader is DeploymentUtils {
...
+            // this test suite needs to run independently so we load a fork here
+            forkId = vm.createFork(vm.envString("ARBITRUM_MAINNET_RPC"));
+            vm.selectFork(forkId);
+            // Deploy contracts
+            vm.startPrank(owner);
+            ArbitrumMainnetDeployer.DeploymentResult memory result =
+                ArbitrumMainnetDeployer.deployAndSetupProtocol(owner);
+            DeploymentUtils.exportDeployment(
+                vm,
+                "collar_protocol_deployment",
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/DeploymentLoader.sol:49]: should indicate that it's a fork?

File: script/arbitrum-mainnet/deployer.sol [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            underlying: IERC20(WBTC),
+            oracleFeeTier: oracleFeeTier,
+            swapFeeTier: swapFeeTier,
+            twapWindow: twapWindow,
+            swapRouter: swapRouterAddress
+        });
+
+        assetPairContracts[2] = DeploymentHelper.deployContractPair(configHub, USDCWBTCPairConfig, owner);
+
+        // DeploymentHelper.PairConfig memory USDCMATICPairConfig = DeploymentHelper.PairConfig({
```
Comment (artdgn) [inline script/arbitrum-mainnet/deployer.sol:142]: should we remove these?

**Review by artdgn — APPROVED** [review]
Not 100% following what's going on in these parts for now, but LGTM.

File: script/arbitrum-mainnet/deploy-contracts.s.sol [hunk 6]
```diff
@@ -3,173 +3,27 @@ pragma solidity 0.8.22;
...
-        for (uint i = 0; i < assetPairContracts.length; i++) {
-            SetupHelper.setupContractPair(configHub, assetPairContracts[i]);
-        }
         vm.stopBroadcast();
 
-        exportDeployment(
-            "collar_protocol_deployment", address(configHub), swapRouterAddress, assetPairContracts
+        DeploymentUtils.exportDeployment(
+            vm,
+            "collar_protocol_deployment",
```
[the remark this answers is not in the document]
Comment (carlosdimatteo) [reply script/arbitrum-mainnet/deploy-contracts.s.sol:22]: it goes into a folder for each chain ID

**Pull request merged**